go-controller/pkg/ovn/pods.go

```diff
@@ -331,12 +331,15 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 		if err != nil {
 			return err
 		}
-		marshalledAnnotation, err := util.MarshalPodAnnotation(&util.PodAnnotation{
-			IPs:      []*net.IPNet{podCIDR},
-			MAC:      podMac,
-			Gateways: []net.IP{gwIP},
-			Routes:   routes,
-		})
+
+		var podAnnotation util.PodAnnotation
+		podAnnotation.IPs = []*net.IPNet{podCIDR}
+		podAnnotation.MAC = podMac
+		podAnnotation.Routes = routes
+		if gwIP != nil {
+			podAnnotation.Gateways = []net.IP{gwIP}
+		}
```

Comment:
The current code is temporary; eventually `getRoutesGatewayIP` will return an array of gateway IPs instead of just a single gateway IP. So I think it would be better to keep the old style/layout (using `&util.PodAnnotation{...`) but just create a `gwIPs` variable before that which is either `[]net.IP{gwIP}` or `nil`

Reply:
changed